xbmc/peripherals/bus/android/PeripheralBusAndroid.cpp

```diff
@@ -339,13 +339,14 @@ bool CPeripheralBusAndroid::GetDeviceId(const std::string& deviceLocation, int&
 bool CPeripheralBusAndroid::ConvertToPeripheralScanResult(const CJNIViewInputDevice& inputDevice, PeripheralScanResult& peripheralScanResult)
 {
   int deviceId = inputDevice.getId();
+  int sources = inputDevice.getSources();
   std::string deviceName = inputDevice.getName();
   if (inputDevice.isVirtual())
   {
     CLog::Log(LOGDEBUG, "CPeripheralBusAndroid: ignoring virtual input device \"%s\" with ID %d", deviceName.c_str(), deviceId);
     return false;
   }
-  if (!inputDevice.supportsSource(CJNIViewInputDevice::SOURCE_JOYSTICK) && !inputDevice.supportsSource(CJNIViewInputDevice::SOURCE_GAMEPAD))
+  if (!((sources & inputDevice.SOURCE_GAMEPAD)==inputDevice.SOURCE_GAMEPAD) && !((sources & inputDevice.SOURCE_JOYSTICK)==inputDevice.SOURCE_JOYSTICK))
```

Comment:
Please use `CJNIViewInputDevice::SOURCE_FOO` instead of `inputDevice.SOURCE_FOO`.